HWB.Rmd

```diff
@@ -0,0 +1,144 @@
+---
+title: "HW5B_Angela_Chun"
+author: "Angela Chun"
+date: "8/3/2020"
+output: github_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(tidyr)
+library(dplyr)
+library(readr)
+library(ggplot2)
+library(stringr)
+```
+
+
+# Load the data
+```{r load data}
+AUC_1 <- read.csv("CTRP_files/AUC_1.csv")
+AUC_2 <- read.csv("CTRP_files/AUC_2.csv")
+cancer_cell_line_info <- read.csv("CTRP_files/cancer_cell_line_info.csv")
+compound_info <- read.csv("CTRP_files/compound_info.csv")
+experiment_info <- read.csv("CTRP_files/Experiment_info.csv")
+```
+
+
+# Join the data 
+```{r joined data}
+#joining all the data into one data frame
+Joined_CTRP_data <- AUC_1 %>%
+  bind_rows(AUC_2) %>%
+  inner_join(experiment_info, by = c("experiment_id" = "expt_id")) %>%
+  inner_join(cancer_cell_line_info, by = "master_ccl_id")%>%
+  inner_join(compound_info, by = "master_cpd_id") 
+
+```
+
+
+
+
+# Answering Questions
+
+
+### Q1) Which cancer type has the lowest AUC values to the compound "vorinostat"?
+```{r Q1}
+
+#finding the cancer_type with lowest mean AUC
+CTRP_VOR <- Joined_CTRP_data %>%
+  filter(cpd_name == "vorinostat") %>%
+  group_by(cancer_type) %>%
+  summarize(AUC = mean(area_under_curve)) %>%
+  arrange(AUC)
+  
+#visualizing the data 
+CTRP_VOR
+
+#Answer = haematopoietic_and_lymphoid_tissue
+
+```
+
+
+## Q1) Graph
+```{r Q1 Graph}
+
+# creating a bar graph for the average AUC of each cancer type
+ggplot(CTRP_VOR, aes(x = reorder(cancer_type, AUC), y = AUC)) + geom_col(fill = "lightslateblue") + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
+  xlab("Cancer type") +
+  ylab("Average AUC")
+
+
+```
+
+
+
+
+## Q2) Which compound is the prostate cancer cell line 22RV1 most sensitive to? (For 22RV1, which compound has the lowest AUC value?)
+```{r Q2}
+
+CTRP_prostate <- Joined_CTRP_data %>%
+  filter(cancer_type == "prostate") %>%
+  group_by(cpd_name) %>%
+  summarize(AUC = mean(area_under_curve)) %>%
+  arrange(AUC)
+
+#filtering for ccl_name == "22RV1" yielded no results, so instead I just filtered for the prostate cancer type and found the compound with lowest AUC. Note that this isn't really related to a specific cell line, but didn't know what else to do.
+
```

Comment:
I agree, I looked at your full list (the one after 5 combination), yours has 247923 lines; and my full list has 462784 lines.
So I guess there are some data is missing?

Noticed you were using bind_rows, and I used rbind, could that be the problem?

Reply:
Oh wait, in the email, he said "First, there was a mistake when I uploaded the files, particularly AUC_1.csv. I made an accident when I was formatting that file so it contained 20,000 rows instead of the 200,0000 rows it was supposed to contain. I caught this, but when I uploaded the files I still seem to have uploaded the older version. Those 180,000 rows are important as they contain the information on 22RV1 for one of the questions I ask later. "


This must be the reason!